File: frontend/src/types/ocr.ts
export interface OCRBBox {
  x: number;
  y: number;
  w: number;
  h: number;
}

export interface OCRBlock {
  id: string;
  text: string;
  bbox: OCRBBox;
  confidence: number;
}

export interface OCRPageResult {
  page_number: number;
  image_width: number;
  image_height: number;
  page_width: number | null;
  page_height: number | null;
  coordinate_space: string;
  blocks: OCRBlock[];
}

File: backend/app/models/ocr.py
from typing import Optional

from pydantic import BaseModel


class OCRBBox(BaseModel):
    x: float
    y: float
    w: float
    h: float


class OCRBlock(BaseModel):
    id: str
    text: str
    bbox: OCRBBox
    confidence: float


class OCRPageResult(BaseModel):
    page_number: int
    image_width: int
    image_height: int
    page_width: Optional[float] = None
    page_height: Optional[float] = None
    coordinate_space: str = "pdf_points"
    blocks: list[OCRBlock]
